libs/@guardian/react-crossword/src/components/Crossword.tsx

```diff
@@ -158,9 +161,8 @@ export const Crossword = ({
 					break;
 				}
 				default: {
-					const upperCaseKey = key.toUpperCase();
-					if (/^[A-Z]$/.test(upperCaseKey) && currentEntryId) {
-						setCellProgress({ ...currentCell, value: upperCaseKey });
+					if (currentEntryId && keyDownRegex.test(key)) {
+						setCellProgress({ ...currentCell, value: key.toUpperCase() });
```

Comment:
trying to make the conditional fail as fast as possible and only perform actions like uppercasing if we need to